server.go

```diff
@@ -238,6 +253,8 @@ type Server struct {
 
 	// A pool for UDP message buffers.
 	udpPool sync.Pool
+
+	quicListener quic.Listener
```

Comment:
same here, i.e. use a net.Listener and check if the type implements quic and then do quic things?

Reply:
`quic.Listener` doesn't implement `net.Listener`.

Reply:
sad - that will imply that a std lib impl will have a hard time as well. Wonder if anyone actually works on that